Test/ActivityPub.Types.Tests/Integration/Serialization/ComplexObjectSerializationTests.cs

```diff
@@ -0,0 +1,90 @@
+            HRef = "https://peer.example/actor",
+        };
+        var target = new ASLink()
+        {
+            HRef = "https://home.example/actor",
+        };
+        var follow = new FollowActivity()
+        {
+            Id = "https://peer.example/actor/activities/1",
+            Object = new LinkableList<ASObject>() { follower },
+            Target = new LinkableList<ASObject>() { target },
+        };
+        ObjectUnderTest = new AcceptActivity()
+        {
+            Actor = new LinkableList<ASObject>() { target },
+            Object = new LinkableList<ASObject>() { (ASObject)follow },
+        };
+        // Accept
+        JsonUnderTest.Should().HaveStringProperty("type", "Accept");
+        JsonUnderTest.Should().HaveProperty("object");
+        JsonUnderTest.GetProperty("object").Should().HaveStringProperty("type", "Follow");
+        
+        // Follow
+        JsonUnderTest.GetProperty("object").Should().HaveProperty("object");
+        JsonUnderTest.GetProperty("object").GetProperty("object").Should().BeJsonString(follower.HRef);
+        JsonUnderTest.GetProperty("object").Should().HaveProperty("target");
+        JsonUnderTest.GetProperty("object").GetProperty("target").Should().BeJsonString(target.HRef);
+    }
+    
+    [Fact(Skip = "Collections not serializing properly")]
+    public void AddNoteToCollectionTest()
+    {
+        var actor = new ASLink()
+        {
+            HRef = "https://home.example/actor",
+        };
+        var note = new NoteObject()
+        {
+            Id = "https://peer.example/actor/activities/1",
+            Content = new NaturalLanguageString("This is a note"),
+        };
+        var collection = new ASCollection()
+        {
+            Id = "https://home.example/actor/collections/1",
+            Items = new LinkableList<ASObject>(new ASLink[]
+            {
+                new(){HRef = "https://home.example/item/whatever"},
+                new(){HRef = "https://home.example/item/whatever-2"}
+            }),
+            TotalItems = 2
+        };
+        ObjectUnderTest = new AddActivity()
+        {
+            Actor = new LinkableList<ASObject>() { actor },
+            Object = new LinkableList<ASObject>() { (ASObject)note },
+            Target = new LinkableList<ASObject>(collection)
+        };
+        
+        JsonUnderTest.Should().HaveStringProperty("type", "Add");
+        JsonUnderTest.Should().HaveProperty("object");
+        JsonUnderTest.GetProperty("object").Should().HaveStringProperty("type", "Note");
+        // Fails here, Collections are serializing as an array. Should be an object with an items property.
+        JsonUnderTest.GetProperty("target").Should().HaveStringProperty("type", "Collection");
+        JsonUnderTest.GetProperty("target").Should().HaveStringProperty("id", "https://home.example/actor/collections/1");
+        JsonUnderTest.GetProperty("target").GetProperty("items").Should().BeJsonObject(); // Really an array
```

Comment:
Collections don't seem to be serializing properly. This test object serializes as
```
{
  "type": "Add",
  "@context": "https://www.w3.org/ns/activitystreams",
  "actor": "https://home.example/actor",
  "object": {
    "type": "Note",
    "@context": "https://www.w3.org/ns/activitystreams",
    "id": "https://peer.example/actor/activities/1",
    "content": "This is a note"
  },
  "target": [
    "https://example.com/whatever",
    "https://example.com/whatever-2"
  ]
}
```

I expected this
```
{
  "type": "Add",
  "@context": "https://www.w3.org/ns/activitystreams",
  "actor": "https://home.example/actor",
  "object": {
    "type": "Note",
    "@context": "https://www.w3.org/ns/activitystreams",
    "id": "https://peer.example/actor/activities/1",
    "content": "This is a note"
  },
  "target": {
    "items": [
      "https://example.com/whatever",
      "https://example.com/whatever-2"
    ],
    "id": "https://home.example/actor/collections/1",
    "type": "Collection"
  }
}
```

Reply:
To get the expected output, the property type will need to be ASCollection instead of LinkableList. Or, I suppose you could put a collection in the list, and then put items in the collection. What does the spec say about this? If it should be a collection, then let's change the property type. But if this is a Mastodon Special, then maybe the collection-in-a-list would be better?

Reply:
The spec doesn't say anything special about the Target property. I've gotten in the habit of assuming that everything that can be a list is a list, because it removes a lot of edge cases. But maybe I shouldn't be?

Reply:
I think it can be a list (because pretty much everything can be), but if it's a list then there should *not* be a `Collection` object. Lists map directly to JSON arrays, while collections have the whole object thing.

Reply:
ohhh nevermind I read the code backwards :woman_facepalming:

Reply:
I was looking at `Object` - which has no collection - instead of `Target` - which does. I'll take a look, there should definitely be an object there. I'm pretty sure I know the issue too.